source/javascripts/core/models/BitriseYmlService.spec.ts

```diff
@@ -1711,6 +1711,107 @@ describe('BitriseYmlService', () => {
     });
   });
 
+  describe('removeWorkflowFromPipeline', () => {
+    it('should remove the workflow from the given pipeline', () => {
+      const sourceYml: BitriseYml = {
+        format_version: '',
+        pipelines: {
+          pl1: {
+            workflows: {
+              wf1: {},
+              wf2: { depends_on: ['wf1'] },
+            },
+          },
+          pl2: {
+            workflows: {
+              wf1: { depends_on: ['wf2'] },
+              wf2: {},
+            },
+          },
+        },
+      };
+
+      const expectedYml: BitriseYml = {
+        format_version: '',
+        pipelines: {
+          pl1: {
+            workflows: {
+              wf2: {},
+            },
+          },
+          pl2: {
+            workflows: {
+              wf1: { depends_on: ['wf2'] },
+              wf2: {},
+            },
+          },
+        },
+      };
+
+      const actualYml = BitriseYmlService.removeWorkflowFromPipeline('pl1', 'wf1', sourceYml);
+
+      expect(actualYml).toMatchBitriseYml(expectedYml);
+    });
+
+    it('should return the original YML if the pipeline does not exist', () => {
+      const sourceAndExpectedYml: BitriseYml = {
+        format_version: '',
+        pipelines: {
+          pl1: {
+            workflows: {
+              wf1: {},
+            },
+          },
+        },
+      };
+
+      const actualYml = BitriseYmlService.removeWorkflowFromPipeline('pl2', 'wf1', sourceAndExpectedYml);
+
+      expect(actualYml).toMatchBitriseYml(sourceAndExpectedYml);
+    });
+
+    it('should return the original YML if the workflow does not exist', () => {
+      const sourceAndExpectedYml: BitriseYml = {
+        format_version: '',
+        pipelines: {
+          pl1: {
+            workflows: {
+              wf1: {},
+            },
+          },
+        },
+      };
+
+      const actualYml = BitriseYmlService.removeWorkflowFromPipeline('pl1', 'wf2', sourceAndExpectedYml);
+
+      expect(actualYml).toMatchBitriseYml(sourceAndExpectedYml);
+    });
+
+    it('should remove the workflows field if it becomes empty after removal', () => {
```

Comment:
Well, I just learned about this one. IMO, if a pipeline has no stages and no workflows either, we can always say that it's an empty graph pipeline. (We probably want to direct people to the graph pipeline anyway.)

Reply:
That's a very good idea and would be much better because we wouldn't need to remember the convention, but unfortunately, a pipeline without a stage or workflow is not valid. Couldn’t this cause a problem?
![Screenshot 2024-11-11 at 09 32 37](https://github.com/user-attachments/assets/3ca3b11a-cc71-4c97-8926-fbac55e6c023)